flixel/tile/FlxBaseTilemap.hx

```diff
@@ -1296,4 +1371,28 @@ abstract FlxTilemapDiagonalPolicy(Int)
 	 * Diagonal movement costs one more than orthogonal movement
 	 */
 	var WIDE = 2;
-}
\ No newline at end of file
+}
+
+enum FlxTilemapPathingMethod
+{
+	/**
+	 * Basic method is calculating distances using an array covering the whole map (was original solution)
+	*/
+	BASIC;
+	/**
+	 * Optimization of basic method where distances are only calculated in a zone around start/end tiles
+	 * - scale > 1.0 determines how this rectangular zone is grown around the rectangle formed by start-end points
+	 * - minHorizonSize is used if any the scaled zone dimension is too small
+	*/
+	BASIC_SCALED_HORIZON(scale:Float, minHorizonSize:Int);
+}
+
+typedef FlxTilemapZone =
```

Comment:
Typedefs can generate slow Dynamic code by the C++ compiler, so we tend to avoid them in HaxeFlixel as much as possible. It would be better if you created a pooled class, look at ```FlxRect``` or ```FlxPoint``` for implementation examples.